### ISSUE TITLE
New rule: Spacing for function params

### ISSUE THREAD

**Issue by rgpass** [jscs-dev/node-jscs#1954]
Goal:

1) Get these two to pass

```javascript
myFunc( spaceRequired );

function myFunc(isSpaceReqd) {}
```

2) Get these two to fail

```javascript
myFunc(spaceRequired);

function myFunc( isSpaceReqd ) {}
```

Essentially want spacing to be required when calling functions, but not when defining them. Currently there is no way to get both of these to pass at the same time.

**Comment by gibson042** [comment]
This duplicates #812, and could be solved by either implementing the sibling rules mentioned in #1254 or  more work along the lines of #1777. It is my impression that [version 3.0 will be closer to the latter](https://github.com/jscs-dev/node-jscs/pull/1777#issuecomment-139855122),  which is why I haven't done the former.

/cc @markelog

**Comment by rgpass** [comment]
I'm having trouble following along with #1254 and am unable to get it to work.

```javascript
// Attempt 1
"disallowSpacesInsideParentheses": false,
"disallowSpacesInsideParenthesizedExpression": false,
"requireSpacesInsideParentheses": {
    "all": true,
    "except": [ "{", "}" ]
},
"requireSpacesInsideParenthesizedExpression": {
    "allExcept": [ "function" ]
}
```

```javascript
// Attempt 2 -- same as above except:
"requireSpacesInsideParenthesizedExpression": {
    "allExcept": [ "function", "ParenthesizedParameters", "FunctionParameters", "Parameters" ]
}
```

**Comment by gibson042** [comment]
To clarify, I don't think existing rules are sufficient to cover this case. There are at least two possible avenues for remedying that:
* Introduce new **SpacesInsideParenthesized…** rules (following #1254)
* Introduce new `ignoreParenthesized…` options in **{Require|Disallow}SpacesInsideParentheses** rules (following #1777)

**Comment by markelog** [comment]
It seems we are moving in both directions with focus on "new rules" approach.

If someone would like to add them, that would be great

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
